solr/core/src/java/org/apache/solr/search/join/XCJFQuery.java

```diff
@@ -261,7 +261,7 @@ private TupleStream createSolrStream() {
     }
 
     private DocSet getDocSet() throws IOException {
-      SolrClientCache solrClientCache = new SolrClientCache();
+      SolrClientCache solrClientCache = searcher.getCore().getCoreContainer().getSolrClientCache();
```

Comment:
Need the `solrClientCache.close();` further down in the method be removed since a shared cache is now used?

Reply:
Yes, good catch!